Dockerfile

```diff
@@ -1,29 +1,47 @@
-# Build the manager binary
-FROM golang:1.17 as builder
+# syntax=mcr.microsoft.com/oss/moby/dockerfile:1.3.1
 
+# Build the manager binary
+FROM golang:1.17 AS builder
+RUN apt-get update && apt-get install -y --no-install-recommends ca-certificates
 WORKDIR /workspace
 # Copy the Go Modules manifests
 COPY go.mod go.mod
 COPY go.sum go.sum
 # cache deps before building and copying source so that we don't need to re-download as much
 # and so that source changes don't invalidate our downloaded layer
-RUN go mod download
+ENV GOCACHE=/root/gocache
+ENV CGO_ENABLED=0
+RUN \
+    --mount=type=cache,target=${GOCACHE} \
+    --mount=type=cache,target=/go/pkg/mod \
+    go mod download
+COPY . .
 
-# Copy the go source
-COPY main.go main.go
-COPY api/ api/
-COPY controllers/ controllers/
+FROM builder AS manager-build
+RUN \
+    --mount=type=cache,target=${GOCACHE} \
+    --mount=type=cache,target=/go/pkg/mod \
+    GOOS=linux GOARCH=amd64 go build -o out/manager main.go
 
-# Build
+FROM builder AS eraser-build
 RUN \
-    --mount=type=cache,target=/root/gocache \
-    GOCACHE=/root/gocache CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -a -o manager main.go
+    --mount=type=cache,target=${GOCACHE} \
+    --mount=type=cache,target=/go/pkg/mod \
+    GOOS=linux GOARCH=amd64 go build -ldflags '-w -extldflags "-static"' -o out/removal ./pkg/eraser
+
+FROM scratch as eraser
```

Comment:
can we just use distroless/static:nonroot here? so we don't have to worry about updating/copying certs

Reply:
Fixed